### PR TITLE
Add missing atomic_fence intrinsics as nops

### LINKED ISSUE #972: Support atomic_fence instrinstic
I'd love to try `miri` on [`evmap`](https://github.com/jonhoo/rust-evmap), but sadly I get the following error:

```
error[E0080]: Miri evaluation error: unimplemented intrinsic: atomic_fence
    --> /home/jon/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/src/libcore/sync/atomic.rs:2412:23
     |
2412 |             SeqCst => intrinsics::atomic_fence(),
     |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^ Miri evaluation error: unimplemented intrinsic: atomic_fence
     |
note: inside call to `std::sync::atomic::fence` at src/write.rs:306:9
    --> src/write.rs:306:9
     |
306  |         atomic::fence(atomic::Ordering::SeqCst);
     |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

Is this a known issue? Some cursory searching did not yield anything that seemed relevant.

### PR DESCRIPTION
**Pull request by nico-abram**
Fixes #972

### CODE REVIEW

**Comment by oli-obk** [comment]
Please add a test calling the stable function from std::sync, so we have some code actually invoking these intrinsics

**Comment by nico-abram** [comment]
Would adding a function like atomic_fences to https://github.com/nico-abram/miri/blob/master/tests/run-pass/atomic.rs using https://doc.rust-lang.org/std/sync/atomic/fn.fence.html with all Orderings be enough?

**Comment by oli-obk** [comment]
Yes, definitely

**Comment by nico-abram** [comment]
Force pushed, fence(Relaxed) panics so trying to test that was a bad idea

File: tests/run-pass/atomic.rs [hunk 1]
```diff
@@ -57,6 +58,15 @@ fn atomic_u64() {
...
+        ATOMIC.compare_exchange_weak(1, 0x100, AcqRel, Acquire),
+        Ok(1)
+    );
     assert_eq!(ATOMIC.load(Relaxed), 0x100);
 }
+
+fn atomic_fences() {
+    fence(SeqCst);
+    fence(Release);
+    fence(Acquire);
```
Comment (RalfJung) [inline tests/run-pass/atomic.rs]: This tests 3 fences but you added 4 intrinsics... I will add a test for the `AcqRel` ordering.

**Pull request merged**